helix-term/src/ui/editor.rs

```diff
@@ -1192,6 +1194,13 @@ impl Component for EditorView {
                     (Mode::Insert, Mode::Normal) => {
                         // if exiting insert mode, remove completion
                         self.completion = None;
+                        // TODO: Use an on_mode_change hook to remove signature help
+                        context.jobs.callback(async {
+                            let call: job::Callback = Box::new(|_editor, compositor| {
+                                compositor.remove(SignatureHelp::ID);
+                            });
+                            Ok(call)
+                        });
```

Comment:
It seems like <kbd>Escape</kbd> is closing the signature help pane but not exiting insert mode. Is that something that would be solved by doing this with a mode change hook?

Reply:
It would be much better to close popup and exit insert mode on `Escape`

Reply:
Yeah that should happen, we do the same for autocompletion